File: src/test/java/pages/CheckoutCompletePage.java
package pages;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.WaitForSelectorState;
/**
 * Page Object Model for Swag Labs Checkout Complete Page (Finish)
 * URL: https://www.saucedemo.com/checkout-complete.html
 */
public class CheckoutCompletePage {
    private final Page page;
    // Locators using provided XPaths
    private final String checkoutCompleteContainer = "//div[@id='checkout_complete_container']";
    private final String thankYouMessage = "//h2[text()='Thank you for your order!']";
    private final String ponyExpressLogo = "//img[@class='pony_express']";
    private final String backHomeButton = "//button[@id='back-to-products']";
    private final String checkoutCompleteTitle = "//span[text()='Checkout: Complete!']";
    public CheckoutCompletePage(Page page) {
        this.page = page;
    }
    /**
     * Wait for Checkout Complete page to load
     */
    public void waitForCheckoutCompletePageToLoad() {
        page.locator(checkoutCompleteTitle).waitFor(new Page.Locator.WaitForOptions()
            .setState(WaitForSelectorState.VISIBLE)
            .setTimeout(10000));
        page.locator(checkoutCompleteContainer).waitFor(new Page.Locator.WaitForOptions()
            .setState(WaitForSelectorState.VISIBLE));
    }
    /**
     * Check if Checkout Complete page is displayed
     * @return true if Checkout Complete page is displayed
     */
    public boolean isCheckoutCompletePageDisplayed() {
        return page.locator(checkoutCompleteTitle).isVisible() && 
               page.locator(checkoutCompleteContainer).isVisible();
    }
    /**
     * Check if 'Thank you for your order!' message is displayed
     * @return true if thank you message is displayed
     */
    public boolean isThankYouMessageDisplayed() {
        page.locator(thankYouMessage).waitFor(new Page.Locator.WaitForOptions()
            .setState(WaitForSelectorState.VISIBLE)
            .setTimeout(5000));
        return page.locator(thankYouMessage).isVisible();
    }
    /**
     * Check if Pony Express logo is displayed
     * @return true if Pony Express logo is displayed
     */
    public boolean isPonyExpressLogoDisplayed() {
        try {
            return page.locator(ponyExpressLogo).isVisible();
        } catch (Exception e) {
            return false;
        }
    }
    /**
     * Click Back Home button
     */
    public void clickBackHomeButton() {
        page.locator(backHomeButton).waitFor(new Page.Locator.WaitForOptions()
            .setState(WaitForSelectorState.VISIBLE));
        page.locator(backHomeButton).click();
        page.waitForLoadState();
    }
}

File: src/pages/CheckoutCompletePage.ts
/**
 * Page Object Model for Checkout Complete (Finish) Page
 * Contains locators and methods for Checkout Complete page interactions
 */
import { WebDriver, By } from 'selenium-webdriver';
import { BasePage } from './BasePage';
export class CheckoutCompletePage extends BasePage {
    // Locators using XPath from provided list
    private checkoutCompleteContainer = By.xpath("//div[@id='checkout_complete_container']");
    private thankYouMessage = By.xpath("//h2");
    private backHomeButton = By.xpath("//button[@id='back-to-products']");
    private ponyExpressImage = By.xpath("//img[@class='pony_express']");
    constructor(driver: WebDriver) {
        super(driver);
    }
    /**
     * Wait for Finish page to load
     */
    async waitForFinishPageToLoad(): Promise<void> {
        await this.waitForElementVisible(this.checkoutCompleteContainer);
        await this.waitForElementVisible(this.thankYouMessage);
        console.log('Finish page loaded successfully');
    }
    /**
     * Verify Finish page loads successfully
     */
    async verifyFinishPageDisplayed(): Promise<void> {
        const isVisible = await this.isElementVisible(this.checkoutCompleteContainer);
        if (!isVisible) {
            throw new Error('Finish page not displayed');
        }
        this.logVerification('Finish page loads successfully');
    }
    /**
     * Get the thank you message text
     */
    async getThankYouMessage(): Promise<string> {
        await this.waitForElementVisible(this.thankYouMessage);
        const message = await this.getText(this.thankYouMessage);
        console.log(`Thank you message: ${message}`);
        return message;
    }
    /**
     * Verify 'Thank you for your order!' message displays
     */
    async verifyThankYouMessageDisplayed(): Promise<void> {
        const message = await this.getThankYouMessage();
        if (!message.includes('Thank you for your order!')) {
            throw new Error(`Expected message not found. Actual: ${message}`);
        }
        this.logVerification("'Thank you for your order!' message displays");
    }
    /**
     * Verify Pony Express Sauce Labs logo displays
     */
    async verifyPonyExpressLogoDisplayed(): Promise<void> {
        const isVisible = await this.isElementVisible(this.ponyExpressImage);
        if (!isVisible) {
            throw new Error('Pony Express logo not displayed');
        }
        this.logVerification('Pony Express Sauce Labs logo displays');
    }
    /**
     * Verify success message and logo display
     */
    async verifyOrderCompletionConfirmed(): Promise<void> {
        await this.verifyThankYouMessageDisplayed();
        await this.verifyPonyExpressLogoDisplayed();
        console.log('Order completion confirmed - Message and Logo displayed');
    }
}